### PR TITLE
HPCC-18528 initial check-in

### PR DESCRIPTION
**Pull request by xwang2713**
Add ECL syntax check.
Return WUID and number of result.

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexi.com> 
@kunalaswani please review Javascript/TypeScript code
@g-pan please review README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,149 @@
-# jupyter-itsecl
-ECL jupyter kernel implementation with typescript based on jupyter-jpecl-kernel
+# ITSECL
+
+ITSECL is a Typescript/ECL kernel for the [Jupyter notebook](http://jupyter.org/)...
+It provide a prototype implementation of HPCC ECL for Jupyter Notebook by referencing [ITypeScript](https://github.com/nearbydelta/itypescript)
```
Comment (g-pan) [inline README.md]: It provides

File: README.md [hunk 2]
```diff
@@ -1,2 +1,149 @@
...
+sudo pip3 install jupyter
+sudo curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+sudo apt-get install -y nodejs
+
+```
+On ubuntu 18.04 replace python3.5-dev with python3.6-dev
+
+Install ITSECL
+### Install ITSECL globally
+You don't need use "sudo" but usually required to configure npm default directory
```
Comment (g-pan) [inline README.md]: You don't need to use "sudo" but it is usually required to configure the npm default directory

File: README.md [hunk 3]
```diff
@@ -1,2 +1,149 @@
...
+
+Install itsecl kernel
+```sh
+itsecl --ts-install=global
+
+# If you want to see itsecl kernel debug message add "--ts-debug"
+# To verify ITSECL kernel (jsecl) installed in jupyter:
+jupyter kernelspec list. "tsecl" is the kernel name.
+```
+To run ITSECL in jupyter notebook in your Jupyter Notebook work directory which can be any directory. The Jupyter notebook files will be saved here.
```
Comment (g-pan) [inline README.md]: in jupyter notebook (2x)?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,149 @@
...
+```sh
+jupyter notebook
+
+# Or with debug information for tsecl kernel:
+
+itsecl --ts-debug
+# Then select kernel "HPCC ECL - TSECL" from "new"
+```
+To run ITSECL in jupyter console (Currenly support for console is limit. Try to use notebook instead).
```
Comment (g-pan) [inline README.md]: support for console is limited.

**Review by g-pan — COMMENTED** [review]
see comments inline.

**Comment by xwang2713** [comment]
Update README.md based on Greg's comments.
Add docker image usage and build instruction.

**Comment by xwang2713** [comment]
@kunalaswani, this project references [ITypeScript](https://github.com/nearbydelta/itypescript) which is a general Typescript Jupyter Notebook kernel. 

The major part is "class ECLExecutor" in  lib/eclkernels.ts.

**Comment by kunalaswani** [comment]
Initial review, looks good.

**Comment by xwang2713** [comment]
I will need update license file per discussion with Flavio and Creighton.

**Comment by xwang2713** [comment]
@richardkchapman this is ready to merge.  I am working on the next round of improvement: JIRA HPCC-2079 - Allow import local ECL file in Jupyter Notebook

**Pull request merged**